windows/tools/OobSimulator/DelegateCommand.cpp

```diff
@@ -0,0 +1,61 @@
+// clang-format off
+#include "pch.h"
+#include "DelegateCommand.h"
+// clang-format on
+
+using namespace winrt::Windows::Foundation;
+
+namespace winrt::OobSimulator::implementation
+{
+DelegateCommand::DelegateCommand(std::function<void(IInspectable)> const& execute, std::function<bool(IInspectable)> const& canExecute)
+{
+    if (execute == nullptr) {
+        throw winrt::hresult_invalid_argument(L"execute");
+    }
+
+    m_execute = execute;
+    m_canExecute = canExecute;
+}
+
+void
+DelegateCommand::Execute(IInspectable const& parameter)
+{
+    if (m_execute != nullptr) {
```

Comment:
I didn't know that std::function can be compared directly to pointers. I think that it would be more idiomatic to use the std::function::operator bool() check here

```suggestion
    if (m_execute) {

Reply:
This is a great suggestion, thanks